uvicorn/protocols/http/h11_impl.py

```diff
@@ -155,6 +155,23 @@ def _unset_keepalive_if_required(self) -> None:
             self.timeout_keep_alive_task.cancel()
             self.timeout_keep_alive_task = None
 
+    def _should_upgrade(
+        self, headers: List[Tuple[bytes, bytes]], event: H11Event
+    ) -> bool:
```

Comment:
I didn't notice before, but we already have the `headers` attribute at this point. 
```suggestion
    def _should_upgrade(self, event: H11Event) -> bool:
```

Reply:
Yep, decided to pass them around instead. Changed it to use the attribute.